server/src/services/poaching-incident.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable, RequestTimeoutException } from '@nestjs/common';
+import { DatabaseService } from './db.service';
+import { PoachingIncident } from '../entity/poaching-incident.entity';
+import { PoachingIncidentType } from '../entity/poaching-incident-type.entity';
+
+@Injectable()
+export class PoachingIncidentService {
+  constructor(private readonly databaseService: DatabaseService) { }
+
+  async addPoachingIncident(long: number, lat: number, pType: string): Promise<boolean> {
+    const con = await this.databaseService.getConnection();
+    const poachingIncident = new PoachingIncident();
+
+    try {    
+    const poachingIncidentType =  await con.getRepository(PoachingIncidentType).findOne({ type: pType });
+
+    poachingIncident.timestamp = new Date();
+    poachingIncident.longitude = long;
+    poachingIncident.latitude = lat;
+    poachingIncident.type = await poachingIncidentType;
```

Comment:
Await not needed here because you did await poachingIndidentType on line 15

Reply:
Will do